src/libs/actions/IOU.ts

```diff
@@ -3257,31 +3233,33 @@ function updateMoneyRequestDistanceRate(
     const allReports = ReportConnection.getAllReports();
     const transactionThreadReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${transactionThreadReportID}`] ?? null;
     const parentReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${transactionThreadReport?.parentReportID}`] ?? null;
+
+    const transaction = allTransactions?.[`${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`];
+    if (transaction) {
+        lodashSet(transaction, 'comment.customUnit.customUnitRateID', transactionChanges.customUnitRateID);
+        lodashSet(transaction, 'comment.customUnit.defaultP2PRate', null);
+
+        const existingDistanceUnit = transaction?.comment?.customUnit?.distanceUnit;
+        const newDistanceUnit = DistanceRequestUtils.getUpdatedDistanceUnit({transaction, policy});
+
+        // If the distanceUnit is set and the rate is changed to one that has a different unit, mark the merchant as modified to make the distance field pending
+        if (existingDistanceUnit && newDistanceUnit !== existingDistanceUnit) {
+            transactionChanges.merchant = TransactionUtils.getMerchant(transaction);
+        }
+    }
```

Comment:
Do not mutate onyx objects. Is that even necessary? Isn't setting `transactionChanges.merchant` enough?